cmd/swarmctl/service/common.go

```diff
@@ -13,28 +13,63 @@ func getService(ctx context.Context, c api.ControlClient, input string) (*api.Se
 	rg, err := c.GetService(ctx, &api.GetServiceRequest{ServiceID: input})
 	if err != nil {
 		// If any error (including NotFound), ListServices to match via ID prefix and name prefix.
-		rl, err := c.ListServices(ctx,
-			&api.ListServicesRequest{
-				Filters: &api.ListServicesRequest_Filters{
-					NamePrefixes: []string{input},
-				},
-			},
-		)
+		service, err := getServiceByName(ctx, c, input)
```

Comment:
Any time you are trying to resolve a set to single fuzzy item, this is the behavior to use. In psuedocode, it looks like this:

```
results := find(query)

switch len(results) {
case 0: not found
case 1: found
default: ambiguous -> error
}
```

This reduces round trips and ensures that you get the right result. If you don't handle the ambiguous case, you end up with a potential security exploit.